File: php/features.php
<?php
declare(strict_types=1);

// Bored SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BoredFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BoredBaseFeature();
            case "test":
                return new BoredTestFeature();
            default:
                return new BoredBaseFeature();
        }
    }
}

File: py/features.py
# Bored SDK feature factory

from feature.base_feature import BoredBaseFeature
from feature.test_feature import BoredTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BoredBaseFeature(),
        "test": lambda: BoredTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
